includes/discovery/ports/zynos.inc.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/*
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ * @author     LukeKelsall <l.kelsall@b4rn.org.uk>
+ */
+
+//loop through ports
+foreach ($port_stats as $index => $port) {
+    //Logically convert swp ports to human readable
+    //only convert swp ports
+    if (substr($port['ifName'], 0, 3) === 'swp') {
+        $portNum = preg_replace('/[a-z]+/', '', $port['ifName']);
+        if ($portNum < 50) {
+            $portNum++;
+            //Leading 0 for single digits
+            if ($portNum < 10) {
+                $ifDescr = '1/0' . $portNum;
+            } else {
+                $ifDescr = '1/' . $portNum;
+            }
+        }
+        if ($portNum > 63 && $portNum < 128) {
+            $portNum = $portNum - 63;
+            //Leading 0 for single digits
+            if ($portNum < 10) {
+                $ifDescr = '2/0' . $portNum;
+            } else {
+                $ifDescr = '2/' . $portNum;
+            }
+        }
+        if ($portNum > 127 && $portNum < 192) {
+            $portNum = $portNum - 127;
+            //Leading 0 for single digits
+            if ($portNum < 10) {
+                $ifDescr = '3/0' . $portNum;
+            } else {
+                $ifDescr = '3/' . $portNum;
+            }
+        }
+        if ($portNum > 191) {
+            $portNum = $portNum - 191;
+            //Leading 0 for single digits
+            if ($portNum < 10) {
+                $ifDescr = '4/0' . $portNum;
+            } else {
+                $ifDescr = '4/' . $portNum;
+            }
+        }
+    } else {
+        //Set port number to match current
+        $ifDescr = $port['ifName'];
+    }
+    //set the libre ifDescr value to new port name
+    $port_stats[$index]['ifDescr'] = $ifDescr;
+    $port_stats[$index]['ifName'] = $ifDescr;
```

Comment:
It might be worthwhile to only override ifName.  if ifDescr is empty, you could set it to the original ifName value.
ifDescr is the port as described by the os
ifName should be the name of the port as used in cli commands
ifAlias is the human configured description of the port

Reply:
There is also an os definition flag to set which field to prefer for displaying the port name.

Reply:
We tested the OS definition flag but the reason that we changed both ifDescr and ifName was because some areas of libre would still show as swp numbers, The cli uses the 1/1 format and so does the writing on the switch so thats another reson why we wanted to change ifDescr. I will test it again to see which areas use ifDescr and if it is still necessary.

Reply:
After testing it would be best if we can change both ifDescr and ifName so we can completely remove the swp numbers. If you guys don't like us changing ifDescr let me know and I will remove it. Please can someone rerun the workflows now I have added the zeropad config so it can test the pull request.